setup.py

```diff
@@ -310,6 +310,14 @@ def get_long_description():
             "test": [
                 "importlib-resources>=1.1.0",
             ],
+            "h5": [
+                "numpy",
+                "pandas",
+                "tables",
+            ],
+            "persistence": [
+                "numpy",
+            ],
```

Comment:
`configobj` is only used by preferences.  Should it be moved here as a `preferences` specific optional dependency?

Reply:
While I agree with the move, we can't move this abruptly because the installation process for anyone depending on preferences will have to change. 

e.g. Mayavi will be adversely affected by such a change, see its install requirement:
https://github.com/enthought/mayavi/blob/61ef7c0eebbdca474b7cfdf6082bfc17fb5aaf59/mayavi/__init__.py#L10-L19

Currently `pip install mayavi` will bring in `apptools`, which will bring in `configobj`. Mayavi does depend on the preferences module. If we are to move `configobj` from install requirement to optional extras requirement, we need to warn about it much in advanced, and make the necessarily adjustment downstream. 

I think we can add `"preferences": ["configobj"]` here, without removing the one in `install_requires`. This is probably the first step to that migration. e.g. Mayavi will change its install requires from `"apptools"` to  `"apptools[preferences]"`